spec/factories/schools.rb

```diff
@@ -104,11 +104,26 @@
             class: "Placements::School",
             parent: :school, traits: %i[placements] do
       transient do
+        hosting_interests { nil }
+        with_hosting_interest { nil }
         with_school_contact { true }
       end
 
       after(:build) do |school, evaluator|
         create(:school_contact, school:) if evaluator.with_school_contact
+
+        # If user passed in with_hosting_interest explicitly (true/false), respect it
+        if !evaluator.with_hosting_interest.nil?
+          create(:hosting_interest, school:, academic_year: Placements::AcademicYear.current.next) if evaluator.with_hosting_interest
+
+        # If user passed in hosting_interests directly, assume they want to manage it
+        elsif evaluator.hosting_interests.present?
+          evaluator.hosting_interests.each { |hi| school.hosting_interests << hi }
+
+        # If neither is passed, use default behavior
+        else
+          create(:hosting_interest, school:, academic_year: Placements::AcademicYear.current.next)
+        end
```

Comment:
The code executed for what the `!evaluator.with_hosting_interest.nil?` and `else` is the same. 
```suggestion
        if evaluator.hosting_interests.present?
          evaluator.hosting_interests.each { |hi| school.hosting_interests << hi }
        elsif with_hosting_interest
          create(:hosting_interest, school:, academic_year: Placements::AcademicYear.current.next)
        end
```

meaning you can set the default transients to:
```
hosting_interests { [] }
with_hosting_interest { true }
```

Reply:
Awesome; ive put the changes in =)